### DISCUSSION TITLE
bind_value_from disable/enable

### DISCUSSION THREAD

**Discussion by emelbaer** [zauberzeug/nicegui#1041] [Ideas / Feature Requests]
Hey there,

I face the problem, when I have a Input/number/text and apply a "bind_value_from" on a instance;
A "user" change of this ui's is no more possible/useful.

Sure one solution is to use "bind_value" instead of "bind_value_from".

-> Freature Request:
For user/client controllable Inputs, "deactivate" bind_value_from (in binding.active_links e.g. move to binding.inactive_links)
When the Element is enabled (than the user/client can enter a own value)

On the otherside when the element is disabled (no user/client input possible) "restore"/"activate" the binding (move from binding.inactive_links back to binding.active_links)

If this is not an new/missing Feature, I would be glat to hear some solution for such situations... 😄 

Best Regards

**Comment by falkoschindler** [comment]
Hi @emelbaer!
I don't quite understand the use case for enabling/disabling binding. What would be an application?
Any why don't you want to use `bind_value`?
Currently there is no way to disable binding at runtime. But if you really need to implement exactly what you're describing, you might simply use two input elements, one with binding and one without, and show/hide them in turn.

**Comment by emelbaer** [comment]
Hi Falko,thanks for replaying.

A situation like: 

A is a number(input field); 
B is generating a Number e.g. a Slider, but this Element is shared amongst the clients.
C is another number holding Element.

B.bind_value_from(C)
A.bind_value_from(B)


But I need to edit A some time and send the Value e.g. by submit button to C - cause B shouldn’t be changed directly from A change…
My first idea was to disable the input A in cases where it should just be updated from B - and than enable A when it’s time to interact… 
but as I pointed out I was no more able to change A😅

Nevertheless, I solved in the Way you described, having a A‘ which is bind to B and just show the value… and A is free from any bind.

BR,Emel

[edit]: formatting - was using the replay function from eMail (note to everyone: do not use this it’s terrible in formating

**Reply by falkoschindler** [reply]
Sorry, without any line breaks and formatting it's hard to follow your thought. You might want to edit your post to make it more readable for everyone.
But I guess your problem is solved. Glad I could help. 🙂
